### ISSUE TITLE
Numba RandomState overload fails with empty size

### ISSUE THREAD

**Issue by ricardoV94** [numba/numba#8975]
## Reporting a bug

- [x] I have tried using the latest released version of Numba (most recent is
 visible in the change log (https://github.com/numba/numba/blob/main/CHANGE_LOG).
- [x] I have included a self contained code sample to reproduce the problem.
  i.e. it's possible to run as 'python bug.py'.

```python
import numpy as np
import numba

@numba.njit
def uniform():
    return np.random.uniform(0, 1, size=())

uniform()  # numba.core.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
```

**Comment by kc611** [comment]
Thank you @ricardoV94 for reporting the issue, I can confirm the bug.

A simple fix can be to allow this edge case to pass through the same corner case branch as `size=None`, the following lines:

https://github.com/numba/numba/blob/0b85096be667ed817a52f5989bf252bda552b8f1/numba/cpython/randomimpl.py#L679-L682

can be modified to: 

```python
    is_empty_tuple = lambda x :isinstance(x, types.Tuple) and len(x.types) == 0
    if (isinstance(low, (types.Float, types.Integer)) and isinstance(
            high, (types.Float, types.Integer)) and
        (is_nonelike(size) or is_empty_tuple(size))):
        return lambda low, high, size: np.random.uniform(low, high)
```

**Comment by NSchiffmacher** [comment]
Hi all, 

I'm new to open-source development and would like to get into it by contributing to numba. I've been able to reproduce the bug, and I've seen that @masaaldosey in #8997 seems to have worked on a fix, but the PR has been closed. Does this mean that no one is currently working on the issue? Should I basically copy this PR and try to fix the issues mentioned there? 

Thanks for your help 😄
